### PR TITLE
AUTO-765 Add Cloudfront distribution classes

### PR DESCRIPTION
**Pull request by aynsof**
Passing build: https://gaautobots.atlassian.net/builds/browse/AM-AUT62/latest

### CODE REVIEW

**Comment by GypsyBojangles** [comment]
I like nice names D:

**Comment by aynsof** [comment]
Passing systems test: https://gaautobots.atlassian.net/builds/browse/AM-AST61-8

File: amazonia/classes/cf_distribution_unit.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                MaxTTL=cache_behavior.max_ttl,
+                PathPattern=cache_behavior.path_pattern,
+                SmoothStreaming=False
+            )
+
+            self.cache_behaviors.append(created_cache_behavior)
+
+    def add_default_cache_behavior(self, title, cf_distribution_config):
+        """
+        Create Cloudfront DefaultCacheBehavior object
```
Comment (GypsyBojangles) [inline amazonia/classes/cf_distribution_unit.py]: missing docstrings

File: test/unit_tests/test_cf_distribution.py [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+        error_page_path=error_page_path,
+        acm_cert_arn=acm_cert_arn,
+        minimum_protocol_version=minimum_protocol_version,
+        ssl_support_method=ssl_support_method
+    )
+
+    return cf_distribution_config
+
+def create_s3_origin(domain_name='amazonia-elb-bucket.s3.amazonaws.com', origin_id='S3-bucket-id',
+                   is_s3=True, origin_access_identity = 'originaccessid1'):
```
Comment (fuddstar) [inline test/unit_tests/test_cf_distribution.py]: Same as above - consider one argument per line

File: test/unit_tests/test_cf_distribution.py [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+            'is_s3': is_s3,
+            'origin_access_identity': origin_access_identity
+        }
+    )
+
+    return origin
+
+
+def create_custom_origin(domain_name='amazonia-elb-bucket.s3.amazonaws.com', origin_id='S3-bucket-id',
+                        is_s3=False, origin_protocol_policy='https-only', http_port='80', https_port='443',
```
Comment (fuddstar) [inline test/unit_tests/test_cf_distribution.py]: ..One argument per line

File: test/unit_tests/test_cf_distribution.py [hunk 4]
```diff
@@ -0,0 +1,288 @@
+#!/usr/bin/python3
+
+from nose.tools import *
+from amazonia.classes.cf_distribution_unit import CFDistributionUnit
+from amazonia.classes.cf_distribution_config import CFDistributionConfig
+from amazonia.classes.cf_origins_config import CFOriginsConfig
+from amazonia.classes.cf_cache_behavior_config import CFCacheBehavior
+from troposphere import Template, cloudfront
+
+def create_cf_distribution_config(aliases=('wwwelb.ap-southeast-2.elb.amazonaws.com'),
```
Comment (fuddstar) [inline test/unit_tests/test_cf_distribution.py]: Are the brackets necessary?

Comment (aynsof) [inline test/unit_tests/test_cf_distribution.py]: Aliases is a list of strings. I'm passing it as a tuple in this case.

**Comment by fuddstar** [comment]
:+1:

**Pull request merged**